src/App.tsx

```diff
@@ -8,7 +12,81 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+type ActiveModal = {
+  todo: Todo | null,
+  index: number,
+  isActive: boolean,
+};
+
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [isActiveTodoList, setIsActiveTodoList] = useState(false);
```

Comment:
better to make just one state for todos. also make state for search query and sort type.
create functions which will be sort or filter todos depands on search query and sort type state